File: T6. C# CONDITIONAL STATEMENTS AND LOOPS - EXERCISES (Java)/src/P07_CakeIngredients.java
import java.util.Scanner;

public class P07_CakeIngredients {
    public static void main(String[] args) {

        Scanner in = new Scanner(System.in);
        int counter = 0;

        while (true) {

            String input = in.nextLine();

            if (!(input.equals("Bake!"))) {
                System.out.printf("Adding ingredient %s.\n", input);
            } else {
                System.out.printf("Preparing cake with %d ingredients.", counter);
                break;
            }
            counter ++;
        }
    }
}

File: T6. C# CONDITIONAL STATEMENTS AND LOOPS - EXERCISES/P07.CakeIngredients.cs
using System;

namespace P07.CakeIngredients
{
    class Program
    {
        static void Main(string[] args)
        {
            int counter = 0;
            while (true)
            {
                string inputCommand = Console.ReadLine();

                if (!(inputCommand.Equals("Bake!")))
                {
                    Console.WriteLine($"Adding ingredient {inputCommand}.");
                    counter++;
                }
                else if (inputCommand.Equals("Bake!") || counter > 20)
                {
                    break;
                }
            }
            Console.WriteLine($"Preparing cake with {counter} ingredients.");
        }
    }
}
